sdformat_urdf/src/sdformat_urdf.cpp

```diff
@@ -494,21 +494,23 @@ sdformat_urdf::convert_joint(const sdf::Joint & sdf_joint, sdf::Errors & errors)
     case sdf::JointType::PRISMATIC:
       urdf_joint->type = urdf::Joint::PRISMATIC;
       break;
-    case sdf::JointType::INVALID:     // Unsupported: fall through to default
-    case sdf::JointType::BALL:        //  |
+    case sdf::JointType::UNIVERSAL:   // Unsupported: fall through to floating
+    case sdf::JointType::BALL:        //  Will require custom TF publisher
     case sdf::JointType::GEARBOX:     //  |
     case sdf::JointType::REVOLUTE2:   //  |
-    case sdf::JointType::SCREW:       //  |
-    case sdf::JointType::UNIVERSAL:   //  V
+    case sdf::JointType::SCREW:       //  V
+      urdf_joint->type = urdf::Joint::FLOATING;
+      break;
+    case sdf::JointType::INVALID:
     default:
       errors.emplace_back(
         sdf::ErrorCode::STRING_READ,
         "Unsupported joint type on joint [" + sdf_joint.Name() + "]");
       return nullptr;
   }
 
-  if (urdf::Joint::FIXED != urdf_joint->type) {
-    // Add axis info for non-fixed joints
+  if ((urdf::Joint::FIXED != urdf_joint->type) && (urdf::Joint::FLOATING != urdf_joint->type)) {
+    // Add axis info for non-fixed and non-floating joints
     const sdf::JointAxis * sdf_axis = sdf_joint.Axis(0);
```

Comment:
I think it may be worth it adding a check here to make sure `nullptr != sdf_axis` just to be safe and we don't risk accessing a null pointer below.

Reply:
This seems like a good idea.

Reply:
Added the check and a test with a new sdf whose joint has no axis tag

Reply:
Thanks for the new test and the check.

I'm not sure about treating this as an error though. I was looking at the SDF spec and it looks like the `axis` element is [optional](http://sdformat.org/spec?ver=1.9&elem=joint#joint_axis). That's probably because it depends on the joint type. I can see an argument for considering a missing axis an error for joint types that do have axes, but I think that should be handled at the `libsdformat` level, not here.

Is it possible to not return an `nullptr` here, and instead just skip the axis parsing if it's not available?

Reply:
Wasn't aware of that, thanks for catching this! But by filtering out FLOATING and FIXED urdf joints, the sdf mapped counterparts reach that point are only continuous, revolute and prismatic. Those should be having an axis and the parser should fail without it?

Reply:
Ok, let's go with the current implementation, I think it's the SDF spec's fault for not being clear about what to do in this case. I'll see if I open an issue there later, but I don't want to block this on that.